### DISCUSSION TITLE
How to setup for local non-admin install path on CPACK ?

### DISCUSSION THREAD

**Discussion by ahoarau** [wixtoolset/issues#6908] [Questions]
I would like to have an installer without elevated privileges.

ref: https://github.com/wixtoolset/issues/issues/6903

## Bugs

* Which version of WiX are you building with?

3.11.2

* Which version of Visual Studio are you building with (if any)?

Visual Studio 2022, 17.3.3, CMake 3.24.1

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?

Windows 10 latest updates

* Describe the problem and the steps to reproduce it.

Setting `CPACK_WIX_ROOT_FOLDER_ID` to local non-admin path fails with errors on all files:

```
`4\WIX\files.wxs(64) : error LGHT0204 : ICE38: Component CM_CP_my_component.boost_program_options_vc142_mt_x64_1_75.dll installs to user profile. It must use a registry key under HKCU as its KeyPath, not a file.`
```

Note: I filled an issue on the cmake side: https://gitlab.kitware.com/cmake/cmake/-/issues/23907
Maybe you can provide additional info.

* Describe the behavior you expected and how it differed from the actual behavior.

I expect two cases:

* Scenario 1 (fix): Allow to install without admin privileges

* Scenario 2 (ideal): Provide user a local / for all user menu, example:
![image](https://user-images.githubusercontent.com/703240/188691808-b2b2e048-74ca-420b-9f92-76c56b9c5cdd.png)
![image](https://user-images.githubusercontent.com/703240/188691912-2b9810e4-5510-44d0-8b6d-e1d8ef7c4aae.png)

**Comment by robmen** [comment]
Sounds like you are interested in implementing a [dual-purpose MSI](https://learn.microsoft.com/en-us/windows/win32/msi/single-package-authoring).

**Reply by ahoarau** [reply]
Yup. Is that possible with wix + cpack ?

**Reply by robmen** [reply]
It is definitely possible with the WiX Toolset. I don't know anything about CPACK's limitations.
